### PR TITLE
Update the workflow to format 4 (Pangolin V.4)

### PR DESCRIPTION
**Pull request by CEJ-R**
### Problem:

Pangolin was updated to V.4 which is better at classifying some older samples then the version we now use. This format changes some of the output, and comes with new guidelines on uploads.

### Changes:

- New conda env created for Pangolin V.4
- Format changed in gensamupload.py
- Path to new env, and new input flag added to pangolin_qsub.sh

#### Old format:
```
taxon,lineage,conflict,ambiguity_score,scorpio_call,scorpio_support,scorpio_conflict,version,pangolin_version,pangoLEARN_version,pango_version,status,note
14_SE300_DC22-034165,BA.2,0.0,1.0,Omicron (BA.2-like),1.000000,0.000000,PLEARN-v1.2.133,3.1.17,2022-03-22,v1.2.133,passed_qc,scorpio call: Alt alleles 65; Ref alleles 0; Amb alleles 0; Oth alleles 0
14_SE300_DC22-032562,BA.2,0.0,0.9465136422693807,Omicron (BA.2-like),0.876900,0.000000,PLEARN-v1.2.133,3.1.17,2022-03-22,v1.2.133,passed_qc,scorpio call: Alt alleles 57; Ref alleles 0; Amb alleles 8; Oth alleles 0
14_SE300_DC22-032557,BA.2,0.0,1.0,Omicron (BA.2-like),1.000000,0.000000,PLEARN-v1.2.133,3.1.17,2022-03-22,v1.2.133,passed_qc,scorpio call: Alt alleles 65; Ref alleles 0; Amb alleles 0; Oth alleles 0
```

#### New format:
```
taxon,lineage,conflict,ambiguity_score,scorpio_call,scorpio_support,scorpio_conflict,scorpio_notes,version,pangolin_version,scorpio_version,constellation_version,is_designated,qc_status,qc_notes,note
14_SE300_DC22-032118,BA.2,0.0,,Omicron (BA.2-like),1.0,0.0,scorpio call: Alt alleles 65; Ref alleles 0; Amb alleles 0; Oth alleles 0,PUSHER-v1.2.133,4.0.1,0.3.16,v0.1.4,False,pass,Ambiguous_content:0.02,Usher placements: BA.2(1/1)
14_SE300_DC22-032173,BA.2,0.0,,Omicron (BA.2-like),0.98,0.02,scorpio call: Alt alleles 64; Ref alleles 1; Amb alleles 0; Oth alleles 0,PUSHER-v1.2.133,4.0.1,0.3.16,v0.1.4,False,pass,Ambiguous_content:0.02,Usher placements: BA.2(1/1)
14_SE300_DC22-032181,BA.2,0.0,,Omicron (BA.2-like),0.98,0.02,scorpio call: Alt alleles 64; Ref alleles 1; Amb alleles 0; Oth alleles 0,PUSHER-v1.2.133,4.0.1,0.3.16,v0.1.4,False,pass,Ambiguous_content:0.02,Usher placements: BA.2(1/1)
```
### Gensam changes:
- File can now be separated by either comma or tab
- New classification format (4) which is the same as output from pangolin V.4
#### ...pangolin_classification_format4.txt:
```
taxon,lineage,conflict,ambiguity_score,scorpio_call,scorpio_support,scorpio_conflict,scorpio_notes,version,pangolin_version,scorpio_version,constellation_version,is_designated,qc_status,qc_notes,note
SE123456,BA.2,0.0,,Omicron (BA.2-like),0.94,0.0,scorpio call: Alt alleles 61; Ref alleles 0; Amb alleles 4; Oth alleles 0,PUSHER-v1.2.133,4.0.1,0.3.16,v0.1.4,False,pass,Ambiguous_content:0.02,Usher placements: BA.2(2/2)
SE123457,Unassigned,,,,,,,PUSHER-v1.2.133,4.0.1,0.3.16,v0.1.4,False,fail,Ambiguous_content:0.8,
SE123458,BA.1,0.0,,Omicron (BA.1-like),0.75,0.0,scorpio call: Alt alleles 44; Ref alleles 0; Amb alleles 12; Oth alleles 3,PUSHER-v1.2.133,4.0.1,0.3.16,v0.1.4,False,pass,Ambiguous_content:0.05,Usher placements: BA.1(1/1)
```

### CODE REVIEW

**Comment by CEJ-R** [comment]
I haven't managed to find anything more that requires an obvious change to mesh the new Pangolin output into this workflow. Do either of you @sima-r or @pbiology know if I've missed something?

File: qsubs_scripts/pangolin_qsub.sh [hunk 1]
```diff
@@ -13,11 +13,12 @@ FASTAIN=$1
 RUNID=$2
 
 module load anaconda2
-source activate /apps/bio/software/anaconda2/envs/pangolin
+source activate /apps/bio/software/anaconda2/envs/pangolin_v4
```
Comment (pbiology) [inline qsubs_scripts/pangolin_qsub.sh:16]: behöver vi en ny pangolin miljö? Det är integet problem egentligen och har lite emd koden att göra, men varför inte bara uppdatera den existerande?

**Review by pbiology — COMMENTED** [review]
Looks good.
Is there anything in the files `nextseq_cronscript.py` or `weekly_report.py` that is affected by this?

**Comment by CEJ-R** [comment]
From my reading of the code, no. But it could be that I missed something, so thought it best to check with the experts.
As for weekly report, it only seems to grab the first two values, based on this: https://github.com/ClinicalGenomicsGBG/sars-cov-2-typing/blob/6ed35d258996394c86435848d5092299dd1da82b/weekly_report.py#L400
So that _should_ not be affected

**Comment by CEJ-R** [comment]
And the cronjob, all I can find is that it (outside of uploading to CLC and sftp) does reformat into gensam-format and fill-empty: https://github.com/ClinicalGenomicsGBG/sars-cov-2-typing/blob/6ed35d258996394c86435848d5092299dd1da82b/nextseq_cronscript.py#L60
So that too _should_ be unaffected by a change in format, as the separators are still the same.

File: qsubs_scripts/pangolin_qsub.sh (hunk 1 above)
Comment (pbiology) [reply qsubs_scripts/pangolin_qsub.sh:16]: ok, fair enough!

**Pull request merged**